### ISSUE TITLE
Handle Cancellation

### ISSUE THREAD

**Issue by gerryfletch** [tpolecat/natchez-http4s#19]
### Description
Request cancellation doesn't populate the request fields (though `case.exit` is correctly set as `cancelled`), breaking any monitoring filters that use for example `http.url exists`.

### Solution
Courtesy of ChristopherDavenport, `guaranteeCase` could be used to cover the `Canceled` case:
```scala
OptionT {
  routes(req).guaranteeCase{
      case Canceled() => // append request + error fields?
      case Errored(e) => OptionT.liftF(addRequestFields *> addErrorFields(e))
      case Succeeded(fa) => fa.value.flatMap{
         case Some(handler) => addRequestFields *> addResponseFields(handler)
         case None => OptionT.liftF(Applicative[F].unit))
     }
  }
}
```

**Comment by zmccoy** [comment]
Took a shot at the above this morning, PR is linked :)

**Comment by gerryfletch** [comment]
Resolved in #20
